src/main/java/seedu/address/logic/commands/UnconfirmMemberCommand.java

```diff
@@ -0,0 +1,137 @@
+            + "[" + PREFIX_COURSEMATE + " COURSEMATE]\n"
+            + "Example: " + COMMAND_WORD + " CS2103T GROUP "
+            + PREFIX_COURSEMATE + " #1 "
+            + PREFIX_COURSEMATE + " John Doe";
+    public static final String MESSAGE_MEMBERS_NOT_IN_GROUP =
+            "Some of the specified members are not in the group";
+    public static final String MESSAGE_SUCCESSFULLY_CONFIRMED = "Group successfully modified, Name: %1$s\n"
+            + "%2$s status of members have been set to not confirmed!";
+
+    private final Set<QueryableCourseMate> queryableCourseMateSet;
+    private final Name groupName;
+
+    /**
+     * Basic constructor for {@code UnconfirmMemberCommand}.
+     * @param queryableCourseMateSet set containing the queryableCourseMate to be added
+     */
+    public UnconfirmMemberCommand(Name groupName, Set<QueryableCourseMate> queryableCourseMateSet) {
+        requireAllNonNull(groupName, queryableCourseMateSet);
+        this.groupName = groupName;
+        this.queryableCourseMateSet = queryableCourseMateSet;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Set<List<CourseMate>> courseMateSet;
+        List<QueryableCourseMate> queryableCourseMates = new ArrayList<>();
+        for (QueryableCourseMate q : queryableCourseMateSet) {
+            queryableCourseMates.add(q);
+        }
+
+        try {
+            courseMateSet = queryableCourseMateSet
+                    .stream()
+                    .map(model::findCourseMate)
+                    .collect(Collectors.toSet());
+        } catch (CourseMateNotFoundException e) {
+            throw new CommandException(Messages.MESSAGE_MEMBERS_DONT_EXIST, e);
+        }
+
+        Group toModify;
+        try {
+            toModify = model.findGroup(groupName);
+        } catch (GroupNotFoundException e) {
+            throw new CommandException(Messages.MESSAGE_INVALID_GROUP_NAME);
+        }
+
+        Group modifiedGroup = new Group(toModify);
+        System.out.println("init Group: " + modifiedGroup.asUnmodifiableObservableList());
+
+        List<CourseMate> courseMateList = new ArrayList<>();
+        int index = 0;
+        for (List<CourseMate> courseMateUnconfirmList: courseMateSet) {
+            //If there are more than 1 matching name
+            if (courseMateUnconfirmList.size() > 1) {
+                return new SimilarNameCommand(queryableCourseMates.get(index)).execute(model);
+            }
+            courseMateList.add(courseMateUnconfirmList.get(0));
+            index += 1;
+        }
+
+        try {
+            for (CourseMate courseMate: courseMateList) {
+                CourseMate targetCourseMate = modifiedGroup.findCourseMate(courseMate.getName()).get(0);
+                CourseMate editedCourseMate = unconfirmCourseMate(courseMate);
+                modifiedGroup.setCourseMate(targetCourseMate, editedCourseMate);
+            }
```

Comment:
looks like there is quite a bit of code duplication. consider abstracting it out?

Reply:
for example, OOP it a bit more by creating an abstract class and both ConfirmMember and UnconfirmMember inherits from it

Reply:
ok thanks!